components/rich_text_editor/rich_text_editor.vue

```diff
@@ -0,0 +1,250 @@
+
+export default {
+  name: 'DtRichTextEditor',
+
+  components: {
+    EditorContent,
+  },
+
+  props: {
+    /**
+     * Value of the input
+     */
+    value: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input is editable
+     */
+    editable: {
+      type: Boolean,
+      default: true,
+    },
+
+    /**
+     * Descriptive label for the input element
+     */
+    inputAriaLabel: {
+      type: String,
+      required: true,
+    },
+
+    /**
+     * Additional class name for the input element. Only accepts a String value
+     * because this is passed to the editor via options. For multiple classes,
+     * join them into one string, e.g. "d-p8 d-hmx96"
+     */
+    inputClass: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input should receive focus after the component has been
+     * mounted. Either one of `start`, `end`, `all` or a Boolean or a Number.
+     * - `start`  Sets the focus to the beginning of the input
+     * - `end`    Sets the focus to the end of the input
+     * - `all`    Selects the whole contents of the input
+     * - `Number` Sets the focus to a specific position in the input
+     * - `true`   Defaults to `start`
+     * - `false`  Disables autofocus
+     * @values true, false, start, end, all, number
+     */
+    autofocus: {
+      type: [Boolean, String, Number],
+      default: false,
+      validator (autofocus) {
+        if (typeof autofocus === 'string') {
+          return RICH_TEXT_EDITOR_AUTOFOCUS_TYPES.includes(autofocus);
+        }
+        return true;
+      },
+    },
+
+    /**
+     * The output format that the editor uses when emitting the "@input" event.
+     * One of `text`, `json`, `html`. See https://tiptap.dev/guide/output for
+     * examples.
+     * @values text, json, html
+     */
+    outputFormat: {
+      type: String,
+      default: 'text',
+      validator (outputFormat) {
+        return RICH_TEXT_EDITOR_OUTPUT_FORMATS.includes(outputFormat);
+      },
+    },
+  },
+
+  emits: [
+    /**
+     * Editor input event
+     * @event input
+     * @type {String}
+     */
+    'input',
+
+    /**
+     * Event to sync the value with the parent
+     * @event input
+     * @type {String}
+     */
+    'update:value',
+
+    /**
+     * Editor blur event
+     * @event blur
+     * @type {FocusEvent}
+     */
+    'blur',
+
+    /**
+     * Editor focus event
+     * @event focus
+     * @type {FocusEvent}
+     */
+    'focus',
+  ],
+
+  data () {
+    return {
+      editor: null,
+    };
+  },
+
+  computed: {
+    extensions () {
+      // These are the default extensions needed jsut for plain text.
+      const extensions = [Document, Paragraph, Text];
+      // TODO: Add logic for extensions passed as props.
+      return extensions;
+    },
+
+    inputAttrs () {
+      const attrs = {
+        'aria-label': this.inputAriaLabel,
+        'aria-multiline': true,
```

Comment:
Should this always be true or is it possible to have a "single line mode"

Reply:
Whoops missed these comments. For now we only know of the message input use case, so I'd keep this as is.